### DISCUSSION TITLE
Modify existing watchface

### DISCUSSION THREAD

**Discussion by MoonCastle85** [ooflet/Mi-Create#32] [Questions-EN]
Is there a way to upload an existing watchface and modify it? I just want to make slight changes to one of the stock faces MiFitness offers but the only files I see in the folder are .xml and .json. Any way to compile them and add them as a project in this app?

**Comment by ooflet** [comment]
Looks like you have a newer format Xiaomi uploads their stock watchfaces in. Run the unpack.exe executable on the resource.bin file located inside the watchface zip, its located at the install location of Mi Create > compiler > unpack.exe

The resource.bin file is just a compiled watchface, once you run the unpacker it'll convert to .fprj files you can open in Mi Create

I plan to support the source code of the format (xml, json files), but that will take some time.

**Comment by DimitreosIV** [comment]
Hi, I am very new to this so pardon my ignorance. I downloaded V1.01. And started creating my own watchface for xiaomi mi band 9. However all I wanted is to modify an existing one. CyberY. I attach a screenshot. Is there a way to download this watchface from the mifitness app and modify it with Micreate? If so can you explain for someone very stupid and new like me? 
thanks.

![sample](https://github.com/user-attachments/assets/988697c3-8b82-4791-89b1-17a9f4afb34d)

**Reply by ooflet** [reply]
You can download the watchface file by applying for m0tral's development servers, you get access to download existing watchfaces or upload your own conveniently using the Mi Fitness mod app.

If you don't want to apply for the servers, you can fish it out of your Mi Fitness app cache. If you're using an Android phone, Mi Fitness stores download watchfaces in /Android/data/com.xiaomi.wearable/files/WatchFace/some-sort-of-id and in a folder with the watchface ID. Keep in mind that this wont always work, in some versions the path is different or Mi Fitness deletes the watchface file once its done.

You can use the unpacker binary in https://github.com/ooflet/Mi-Create/blob/main/src/compiler/unpack.exe to unpack the watchface binary.

I've downloaded and unpacked the CyberY watchface for you and fixed up some broken images to save you the hassle 
[CyberY.zip](https://github.com/user-attachments/files/18091639/CyberY.zip)

**Reply by DimitreosIV** [reply]
Man you are a hero!! I will download from you but also I will try your instructions so I learn! thank you so much!

**Comment by mrivar** [comment]
How do you apply to m0tral's development servers? I've been looking up online and I cannot seem to find anything 😅 and also cannot see the Mi Fitness app cache. 

I'm looking to download the Shooting Hoops watchface, the one in the pic i attached (but in english version).

EDIT: I was ableto get to the folder `/Android/data/com.xiaomi.wearable/files/WatchFace/some-sort-of-id` by connecting the phone to the pc. While when looking on my phone the folder `/Android/data` was empty, on my pc the `com.xiaomi.wearable/` folder was there.

![image](https://github.com/user-attachments/assets/5c668592-f0f3-482a-b586-b1a344e8f5bd)

**Reply by ooflet** [reply]
Follow the steps in Mi Create's documentation to apply for access to the servers https://ooflet.github.io/docs/quickstart/testing/
